spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/store/MongoDbChannelMessageStore.java

```diff
@@ -96,12 +99,22 @@ public boolean isPriorityEnabled() {
 	@Override
 	public void afterPropertiesSet() {
 		super.afterPropertiesSet();
-		getMongoTemplate()
-				.indexOps(this.collectionName)
-				.ensureIndex(new Index(MessageDocumentFields.GROUP_ID, Sort.Direction.ASC)
-						.on(MessageDocumentFields.PRIORITY, Sort.Direction.DESC)
-						.on(MessageDocumentFields.LAST_MODIFIED_TIME, Sort.Direction.ASC)
-						.on(MessageDocumentFields.SEQUENCE, Sort.Direction.ASC));
+		if (this.getCreateIndex()) {
+			getMongoTemplate()
+					.indexOps(this.collectionName)
+					.ensureIndex(new Index(MessageDocumentFields.GROUP_ID, Sort.Direction.ASC)
+							.on(MessageDocumentFields.PRIORITY, Sort.Direction.DESC)
+							.on(MessageDocumentFields.LAST_MODIFIED_TIME, Sort.Direction.ASC)
+							.on(MessageDocumentFields.SEQUENCE, Sort.Direction.ASC));
+		}
+	}
+
+	public void setCreateIndex(boolean createIndex) {
```

Comment:
This is a `public` API, therefore it needs a Javadoc including `@since 6.0.8`.
Right, I know, we have many missed Javadocs in other places, but let's don't introduce more!